python/ray/experimental/workflow/common.py

```diff
@@ -163,7 +170,24 @@ def result(self) -> WorkflowExecutionResult:
         return self._result
 
     @property
-    def id(self) -> StepID:
+    def _name(self) -> str:
```

Comment:
I thought we were unifying step_id and name. After this PR they should be the same attribute right?

Reply:
no. name is static property, step_id is dynamic property. They are different.

Reply:
For example, the id can be different than the name if there is duplicate